### PR TITLE
OPS-1321 Include/Exclude Drafts from Total Agreement Value

### PR DESCRIPTION
**Pull request by stevtek**
## What changed

* summary cards (Total Budget Lines, Total Agreement Value) moved to budget lines tab
* toggle to include/exclude Drafts added to budget lines tab

## Issue

#1321 

## How to test

* on the agreement details page go the budget lines tab and try the toggle

### With Toggle Off (default) for agreement with just two Draft BLIs
image

### With Toggle On
image

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [ ] ~Form validations updated~

### CODE REVIEW

**Review by xlorepdarkhelm — APPROVED** [review]
LGTM, works as intended.

File: frontend/cypress/e2e/agreementDetails.cy.js [hunk 1]
```diff
@@ -7,18 +7,14 @@ beforeEach(() => {
 });
 
 afterEach(() => {
-    cy.injectAxe();
-    cy.checkA11y(null, null, terminalLog);
+    // cy.injectAxe();
```
Comment (fpigeonjr) [inline frontend/cypress/e2e/agreementDetails.cy.js:10]: oh uh

File: frontend/src/helpers/utils.js [hunk 2]
```diff
@@ -58,6 +58,10 @@ export const formatDateToMonthDayYear = (date) => {
     return new Date(date).toLocaleString("en-US", { month: "long", day: "numeric", year: "numeric" });
 };
 
+// Is there a better place for this?
```
Comment (fpigeonjr) [inline frontend/src/helpers/utils.js:61]: maybe in a `StatusTag.helpers.js` in the same directory as the component unless its used in other places so then `/helpers/tag.helpers.js`

File: frontend/src/helpers/utils.js [hunk 3]
```diff
@@ -58,6 +58,10 @@ export const formatDateToMonthDayYear = (date) => {
...
+// List of BLI statuses which are considered Draft
+export const draftBudgetLineStatuses = ["DRAFT", "UNDER_REVIEW"];
+
```
Comment (fpigeonjr) [inline frontend/src/helpers/utils.js:64]: 🤔 was wondering about this, when the toggle is inactive, the "UNDER_REVIEW" BLIs don't show.

File: frontend/src/pages/agreements/details/AgreementBudgetLines.js [hunk 4]
```diff
@@ -17,16 +22,46 @@ import useAlert from "../../../hooks/use-alert.hooks";
...
+    }, {});
+
     // if there are no BLIS than the user can edit
     if (agreement?.budget_line_items?.length === 0) {
         setIsEditMode(true);
     }
 
     return (
         <CreateBudgetLinesProvider>
+            <AgreementBudgetLinesHeader
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/details/AgreementBudgetLines.js:51]: ✨new Component

File: frontend/src/pages/agreements/details/AgreementBudgetLinesHeader.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        <>
+            <div className="display-flex flex-justify flex-align-center">
+                <h2 className="font-sans-lg">{heading}</h2>
+                <button
+                    id="toggleDraftBLIs"
+                    className="hover:text-underline cursor-pointer"
+                    onClick={() => setIncludeDrafts(!includeDrafts)}
+                >
+                    <FontAwesomeIcon
+                        icon={includeDrafts ? faToggleOn : faToggleOff}
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/details/AgreementBudgetLinesHeader.js:25]: nice

File: frontend/src/pages/agreements/details/AgreementBudgetLinesHeader.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                        size="xl"
+                        className="margin-right-1 cursor-pointer usa-tooltip"
+                        title={includeDrafts ? "On (Drafts included)" : "Off (Drafts excluded)"}
+                        // data-position="top"
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/details/AgreementBudgetLinesHeader.js:29]: commented code here

File: frontend/src/pages/agreements/list/AgreementsList.jsx [hunk 7]
```diff
@@ -139,7 +139,7 @@ export const AgreementsList = () => {
             (filters.budgetLineStatus.draft === true && agreement.budget_line_items.length === 0) ||
             (filters.budgetLineStatus.draft === true &&
                 agreement.budget_line_items.some((bli) => {
-                    return bli.status === "DRAFT" || bli.status === "UNDER_REVIEW";
+                    return draftBudgetLineStatuses.includes(bli.status);
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/list/AgreementsList.jsx:142]: 🥇 great catch here

File: frontend/src/pages/agreements/details/AgreementBudgetLines.js (hunk 4 above)
Comment (fpigeonjr) [reply frontend/src/pages/agreements/details/AgreementBudgetLines.js:51]: maybe it should live in `components/Agreements/AgreementBudgetLinesHeader`

**Review by fpigeonjr — CHANGES_REQUESTED** [review]
💯 Looks great @stevtek
- could we get the text-color and icons colors to match the design?

SCR-20230929-jhyi
SCR-20230929-jiia

File: frontend/src/helpers/utils.js (hunk 2 above)
Comment (stevtek) [reply frontend/src/helpers/utils.js:61]: It is used elsewhere in the agreement list filters and upcoming draft toggles in other places.
I'll leave here for now, but I am starting to wonder if we need another place for biz logic organized by class (Agreement, BudgetLineItem).

File: frontend/cypress/e2e/agreementDetails.cy.js (hunk 1 above)
Comment (stevtek) [reply frontend/cypress/e2e/agreementDetails.cy.js]: foxed

File: frontend/src/components/Agreements/AgreementBudgetLinesHeader.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+                <h2 className="font-sans-lg">{heading}</h2>
+                <button
+                    id="toggleDraftBLIs"
+                    className="hover:text-underline cursor-pointer"
+                    onClick={() => setIncludeDrafts(!includeDrafts)}
+                >
+                    <FontAwesomeIcon
+                        icon={includeDrafts ? faToggleOn : faToggleOff}
+                        size="xl"
+                        className={`margin-right-1 cursor-pointer ${includeDrafts ? "text-primary" : "text-base"}`}
```
Comment (fpigeonjr) [inline frontend/src/components/Agreements/AgreementBudgetLinesHeader.js:27]: 🥇

**Review by fpigeonjr — APPROVED** [review]
thanks Steve, looks great!

**Comment by stevtek** [comment]
@fpigeonjr 
> * could we get the text-color and icons colors to match the design?
I've updated the colors using USWDS classes with the colors in the Figma design

**Pull request merged**